### ISSUE TITLE
Editor Idle Power Consumption and Sleep Prevention

### ISSUE THREAD

**Issue by davidhudson** [godotengine/godot#55608]
### Godot version

3.4 Stable

### System information

Windows 11 / Surface Pro 8 / i7 x86_64

### Issue description

Having the editor open but minimised / unfocused eats up 1-2 hours of battery life. Cause seems to be continuous use of `Windows Audio Device Graph Isolation`. `Desktop Window Manager` usage is also much higher than when the editor is closed.

![image](https://user-images.githubusercontent.com/10351202/144683665-9bae2372-9515-485b-abb4-a583a362b469.png)

This can also be seen by running `powercfg /requests` in the terminal.

```
SYSTEM:
[DRIVER] Realtek High Definition Audio(SST) (INTELAUDIO\FUNC_01&VEN_10EC&DEV_0274&SUBSYS_10EC122E&REV_1000\5&1543d6b8&0&0001)
An audio stream is currently in use.
```

Sometimes, but not consistently this also prevents sleep when closing the keyboard cover.

Might be analogous to these issues affecting MacOS:
https://github.com/godotengine/godot/issues/28039
https://github.com/godotengine/godot/issues/38154

Maybe this is a minor issue in the grand scheme of things, but usability is impacted when you have to keep opening and closing a project when you don't expect to work on it for even a hour less you sacrifice some battery!

### Steps to reproduce

Open the editor

**Comment by Calinou** [comment]
As a workaround, start the editor with the `--audio-driver Dummy` [command line argument](https://docs.godotengine.org/en/stable/getting_started/editor/command_line_tutorial.html).

**Comment by lawnjelly** [comment]
> Start the editor with `--audio-driver Dummy` doesn't work for me. pulseaudio is still using the same amount of CPU. In KDE, I can see there is `speech-dispatcher-dummy` playing audio now.

This is in 3.x or 4.x or both?

This sounds like a new problem related to text to speech creating an audio output. :thinking:

**Comment by wb14123** [comment]
@lawnjelly This is only in 4.x branch. When I try 3.x with `--audio-driver Dummy`, there is no application showing playing sound in KDE.

**Comment by bruvzg** [comment]
`speech-dispatcher` is a completely independent process, so we can't directly control what it is doing.

I guess we can postpone `speech-dispatcher` init until it is used, but since starting it take some time, it will make first TTS call slow, so I'm not sure about it.

Is it causing significant CPU usage, or just listed in the audio mixer?

**Comment by wb14123** [comment]
@bruvzg It is causing significant CPU usage for pulseaudio when using `--audio-driver Dummy`

**Comment by bruvzg** [comment]
> Just to confirm, is there a project setting for text to speech? It should presumably be off by default unless required in the project.

No, there's no option to disable it completely. But if it's causing issues, we probably should add one.

> when using --audio-driver Dummy

TTS is not affected by `audio-driver` option, since sound is generated by the `speech-dispatcher` (or other system component depending on OS) not by Godot.

**Comment by lawnjelly** [comment]
This regression (I'm presuming the text to speech) also appears to be happening in 3.x, I'm getting up to 11 percent CPU used by pulse audio with Godot audio switched off. I'll have a look at fixing this asap, as this is a big problem for low power computers.
